lib/http_requests.py

```diff
@@ -179,21 +177,21 @@ def get_jira_link(data):
 
 
 def get_branch_advisory_ids(branch_name):
+    advisory_data = {}
     if branch_name.split('-')[-1] in ["3.11", "4.5"]:  # versions which do not have releases.yml
         url = f"{os.environ['GITHUB_RAW_CONTENT_URL']}/{branch_name}/group.yml"
         yml_data = get_http_data(url)['advisories']
+        advisory_data = {"current": yml_data, "previous": {}}
+    else:
+        advisories = get_advisories(branch_name)
+        if advisories:
+            for advisory in advisories:
+                try:
+                    jira_link = advisory[2]
+                except IndexError:
+                    jira_link = None  # Assign a default value
+                advisory_data[advisory[0]] = [advisory[1], jira_link]
+        if not advisories:  # If advisories is None or empty
+            return {"current": {}, "previous": {}}  # Return empty data structure
 
-        return {"current": yml_data, "previous": {}}
-
-    advisories = get_advisories(branch_name)
-
-    if advisories:
-        advisory_data = {}
-        for advisory in advisories:
-            try:
-                jira_link = advisory[2]
-            except IndexError:
-                jira_link = None  # Assign a default value
-            advisory_data[advisory[0]] = [advisory[1], jira_link]
-        return advisory_data
-    return {"current": {}, "previous": {}}
+    return advisory_data
```

Comment:
Im curious, what happens if `{}` is returned by `advisory_data`. Will the front end throw an error, or will it be in perpetual loading state? If its the latter, can we say there is an error while loading or something like that?

Reply:
The frontend will display two tables, current and previous with the table simply responding with "No data". I will do some error handling for this scenario being more obvious to the user.

Reply:
well error handling .. is well difficult, mainly because of the multiple renderings of the table. At first the table always renders with no data, which would trigger any error state and show the error message, even if the data is still being fetched. I propose keeping the logic as it is and creating a separate ticket to remake the entire logic of the frontend to prevent multiple rerenderings, making implementing other functions like this into the frontend more simple.